### ISSUE TITLE
non-exhaustive `when` statements generated for `oneof` trigger compiler warnings

### ISSUE THREAD

**Issue by RaphaelTarita** [open-toast/protokt#172]
The class generated when using a `message` with a `oneof` field will contain a `when` statement that is not exhaustive. As the Kotlin compiler mentions it, this code will cause a compilation error from 1.7 onwards:

> Non exhaustive 'when' statements on sealed class/interface will be prohibited in 1.7, add 'null' branch or 'else' branch instead

It seems that `oneof` generates a sealed class, and then two similar `when`-statements are generated in `messageSize()` and `serialize()`. Both use the `oneof` field as the subject, however that field is nullable. Kotlin therefore expects either a branch for the `null` case, or an `else` branch.

A minimal reproducible example would be:
```proto
message Foo {
  oneof bar {
    double a = 1;
    string b = 2;
  }
}
```
The code that protokt generates should trigger the aforementioned compiler warning. A full example would be available in one of my [GitHub repositories](https://github.com/RaphaelTarita/msw-server/blob/develop/src/main/proto/msw/server/rpc/versions/Progress.proto).

Since Kotlin 1.7 is around the door, and the update would essentially break any build involving `oneof` fields, this should probably be fixed soon.

**Comment by andrewparmet** [comment]
Thanks for the report. This was fixed in https://github.com/open-toast/protokt/pull/170 and released in [0.9.0](https://github.com/open-toast/protokt/releases/tag/0.9.0). Please try it out and get back if there's an edge case we missed.

**Comment by RaphaelTarita** [comment]
oh, yeah, my bad. Didn't notice that there was an update available. The update has fixed the problem, however while migrating gRPC, I came across another problem that I reported as a separate issue (#173)
